### PR TITLE
Replace private tag_options call with public content_tag helper

### PR DESCRIPTION
**Pull request by domcleal**
The private `tag_options` method was refactored away in
rails/rails@a65a3bd. Adds ActionView to spec to make all HTML-safe
extensions available.

### CODE REVIEW

File: spec/integration/rails_helper_spec.rb [hunk 1]
```diff
@@ -71,23 +62,29 @@ def tag_options(options)
...
+    should_receive(:content_tag).with(:a, '&#9650;&nbsp;Field', hash_including(:class => 'ascending'))
 
     params[:order] = "field ASC"
     sort("field")
   end
 
   it "should add descending style for current descending sort order " do
     should_receive(:url_for)
-    should_receive(:a_link).with('&#9660;&nbsp;Field', anything, hash_including(:class => 'descending'))
+    should_receive(:content_tag).with(:a, '&#9660;&nbsp;Field', hash_including(:class => 'descending'))
```
Comment (wvanbergen) [inline spec/integration/rails_helper_spec.rb]: Not introduced by your changes, but...

Not a big fan of these tests. Why are we using expectations, rather than asserting the HTML being generated is correct? That is a less fragile and more straightforward approach.

File: Gemfile [hunk 2]
```diff
@@ -1,8 +1,11 @@
 source 'https://rubygems.org'
 gemspec
 
+gem 'actionview'
 gem 'activerecord'
 
+gem 'nokogiri', '~> 1.6.0' if RUBY_VERSION.start_with?('2.0')
```
Comment (wvanbergen) [inline Gemfile]: Why is this necessary?

Also, maybe it's time to drop support for Ruby 2.0 given that it has been EOLed for a while.

File: spec/integration/rails_helper_spec.rb (hunk 1 above)
Comment (domcleal) [inline spec/integration/rails_helper_spec.rb]: Yeah, I think I see your point, though I would be hesitant to match the generated HTML too closely in case the implementation of the tag/content_tag builder changes - even reordering of attributes in the tag could break regex matching. This shouldn't really be a test of `content_tag`.

I can remove expectations and match the generated tag to regular expressions of the HTML, or add a dep on a HTML parser (similar to the Rails test helper `assert_tag`).

File: Gemfile (hunk 2 above)
Comment (domcleal) [inline Gemfile]: The latest version of nokogiri (1.7.x) requires Ruby 2.1 or higher, it's a test-only pin.

File: spec/integration/rails_helper_spec.rb (hunk 1 above)
Comment (wvanbergen) [inline spec/integration/rails_helper_spec.rb]: I personally am happy to just do exact string matching until that becomes a problem. I am pretty sure we can depend on `content_tag` to generate stable HTML. But if you want to do something fancier, feel free :)

File: Gemfile (hunk 2 above)
Comment (wvanbergen) [inline Gemfile]: 👍 Let's leave this here for now in that case. Once we re-evaluate the versions we support, we can 💥 it.

File: spec/integration/rails_helper_spec.rb (hunk 1 above)
Comment (domcleal) [inline spec/integration/rails_helper_spec.rb]: OK, it does seem stable over the three current Rails releases so I've just used an exact match in an additional commit to the PR.
